public/style_editor.js

```diff
@@ -4308,18 +4308,28 @@ class InOutTrLClass extends STYLE {
     this.on_ = false;
     this.out_active_ = false;
     this.in_active_ = false;
+    this.ignitionDetector = new OneshotEffectDetector(EFFECT_IGNITION);
   }
   run(blade) {
     this.OFF.run(blade);
 
     if (this.on_ != blade.is_on()) {
-      this.on_ = blade.is_on();
-      if (this.on_) {
-        this.OUT_TR.begin();
-        this.out_active_ = true;
-      } else {
+      if (!blade.is_on()) {
+        this.on_ = false;
+        this.ignitionDetector.last_detected_ = micros();
+        this.out_active_ = false;
         this.IN_TR.begin();
         this.in_active_ = true;
+      } else {
+        this.on_ = true;
+        this.in_active_ = false;
+      }
+    }
+    // Ignition based on EFFECT_IGNITION
```

Comment:
This seems to be deviation from how InOutTrL works in ProffieOS.
Why? Ideally style editor styles should be as similar as possible to their ProffieOS counterparts.